scripts/named_entity_recognition/code/train_helper.py

```diff
@@ -0,0 +1,179 @@
+# coding: utf-8
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# Some functions are used to aid the training model
+# @author：kenjewu
+# @date：2018/12/12
+
+
+import os
+import logging
+import numpy as np
+from mxnet import autograd, gluon, nd
+
+ROOT = os.path.dirname(os.path.abspath(os.path.expanduser(__file__) + os.path.sep + '..'))
+
+
+def conll_evaluate(output_file, score_file):
+    '''
+    use conll2003evaluate shell to evaluate the perfomance of algorithm
+
+    Args:
+        output_file (str): path of model predict result
+        score_file (str): path of socre result
+
+    Returns:
+        acc (float): accuracy
+        precision (float): precision
+        recall (float): recall
+        f1 (float): F1 score
+    '''
+    code_path = os.path.dirname(os.path.abspath(os.path.expanduser(__file__)))
+    eval_sh = code_path + os.path.sep + 'conll2003eval2.sh < % s > %s'
+
+    os.system(eval_sh % (output_file, score_file))
+    with open(score_file, 'r') as fin:
+        fin.readline()
+        line = fin.readline()
+        fields = line.split(";")
+        acc = float(fields[0].split(":")[1].strip()[:-1])
+        precision = float(fields[1].split(":")[1].strip()[:-1])
+        recall = float(fields[2].split(":")[1].strip()[:-1])
+        f1 = float(fields[3].split(":")[1].strip())
+    return acc, precision, recall, f1
+
+
+def clip_gradients(model, clip, ctx):
```

Comment:
Why not use https://mxnet.incubator.apache.org/api/python/gluon/gluon.html#mxnet.gluon.utils.clip_global_norm ?

Reply:
Yes, this is just my personal habit, because at the beginning there was no such interface, now I will switch to this interface in order to make the code more concise.